main.go

```diff
@@ -120,43 +132,40 @@ func main() {
 		logger.Infof("Error parsing legacy signature support, applying defaults: %v", err)
 		supportLegacySigs = true
 	}
-
-	var authPrivKeyPem string
-	authPrivKeyPemString := envLoader.GetAndLogEnvVar("ROKWIRE_CORE_AUTH_PRIV_KEY", false, true)
-	if authPrivKeyPemString != "" {
-		//make it to be a single line - AWS environemnt variable issue
-		authPrivKeyPem = strings.ReplaceAll(authPrivKeyPemString, `\n`, "\n")
-	} else {
-		authPrivateKeyPath := envLoader.GetAndLogEnvVar("ROKWIRE_CORE_AUTH_PRIV_KEY_PATH", true, false)
-		authPrivKeyPemBytes, err := os.ReadFile(authPrivateKeyPath)
-		if err != nil {
-			logger.Fatalf("Could not find auth priv key file: %v", err)
-		}
-
-		authPrivKeyPem = string(authPrivKeyPemBytes)
+	currentAuthPrivKey := parsePrivKeyFromEnvVar("ROKWIRE_CORE_AUTH_PRIV_KEY", envLoader, supportLegacySigs, logger)
+	if currentAuthPrivKey == nil {
+		logger.Fatalf("Cannot parse the current private key: %v", err)
 	}
 
-	alg := keys.PS256
-	if supportLegacySigs {
-		alg = keys.RS256
-	}
-	authPrivKey, err := keys.NewPrivKey(alg, authPrivKeyPem)
+	oldSupportLegacySigsStr := envLoader.GetAndLogEnvVar("ROKWIRE_CORE_OLD_SUPPORT_LEGACY_SIGNATURES", false, false)
```

Comment:
What is the difference between the new and the old legacy signatures?

Reply:
The difference is to account for the potential to change from a `RS256` key to a `PS256` key. The legacy signature support is used in `parsePrivKeyFromEnvVar` to set the correct algorithm for the key.